presubmit.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/python3
```

Comment:
how do we maintain this, i.e. say we modify the CI. I'd prefer not needing to modify it in 2 places.

is there a way to run the CI script locally that we can use instead?

Reply:
Can’t run GH CI locally without an additional dependency(act) and docker it seems, but we could run this on GH CI instead

Reply:
I just tried `act` and it's super easy: apt-get -ed docker and act, then `sudo act` in the root of this repo.

Let's not do the CI change, and instead use act, otherwise we add work for ourselves wrt having to maintain a CI-like replacement; we also need to figure out stuff like how to handle error output - are we ok to leave it interleaved? (I don't think that'd be very helpful). With act + CI we don't have to worry about any of that.

Reply:
The motivation for this was to have a fast, and local way of running the CI steps, for development. 

act is fine and all but it isn't fast since GH CI is limited to doing steps sequentially, which using a script that spawns its own processes solves. (1m37s vs 32s, without act redownloading packages) So I still think we should have the script, even if using act.

I'm not sure what you mean by interleaved output though, you can see what the output looks with -vvv here https://github.com/Northbadge/ml-compiler-opt/runs/7550324734?check_suite_focus=true having it fail-fast on the CI is also an option instead of having it continue running the other jobs, which replicates what the CI does.

Reply:
If we refactor the Github CI to use multiple jobs (encapsulating each current step in a separate job), that should allow both the Github CI and `act` to run things in parallel. We'll probably be spending a little bit more time in setup (I would imagine even more on the Github CI side), but either way things should be significantly faster and all the steps that would need to be performed locally a pretty cheap (mostly just container creation/maybe copying some files).